components/support/rc_crypto/src/agreement.rs

```diff
@@ -124,6 +124,29 @@ impl PublicKey {
     }
 }
 
+/// An unparsed public key for key agreement.
+pub struct UnparsedPublicKey<'a> {
+    alg: &'static Algorithm,
+    bytes: &'a [u8],
+}
+
+impl<'a> UnparsedPublicKey<'a> {
+    pub fn new(algorithm: &'static Algorithm, bytes: &'a [u8]) -> Self {
+        Self {
```

Comment:
It would be a good idea to assert that the length of `bytes` matches the expected length for the `algorithm`; since so many places are coercing a byte array to the key-type, it'd be good to know if we accidentally typo something in the future.

Reply:
Yup we do that in https://github.com/mozilla/application-services/blob/b46b35d1c45088875cddd904c489aa4d4c0d65b5/components/support/rc_crypto/nss/src/ec.rs#L353-L365.